driver-workload-executor/src/main/com/mongodb/workload/WorkloadExecutor.java

```diff
@@ -16,110 +16,149 @@
+        System.out.println("Path to results directory: '" + pathToResultsDirectory + "'");
+
         Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            System.out.println("Running shutdown hook");
+            terminateLoop = true;
             try {
-                done = true;
-                writeResults(outputDirectory);
-            } catch (IOException e) {
+                if (!terminationLatch.await(1, TimeUnit.MINUTES)) {
+                    System.err.println("Terminating after waiting for 1 minute for results to be written");
+                } else {
+                    System.out.println("Terminating.");
+                }
+            } catch (InterruptedException e) {
                 e.printStackTrace();
             }
         }));
 
-        String connectionString = System.getProperty("org.mongodb.test.uri", "mongodb://localhost");
 
-        MongoClient mongoClient = MongoClients.create(connectionString);
+        BsonDocument fileDocument;
 
-        String workloadSpec = System.getProperty("org.mongodb.workload.spec");
-        if (workloadSpec == null) {
-            throw new IllegalArgumentException("'org.mongodb.workload.spec' system property is required");
+        try (FileReader reader = new FileReader(pathToWorkloadFile)) {
+            fileDocument = new BsonDocumentCodec().decode(new JsonReader(reader), DecoderContext.builder().build());
         }
-        BsonDocument workload = BsonDocument.parse(workloadSpec);
-
-        MongoDatabase database = mongoClient.getDatabase(workload.getString("database").getValue());
-        MongoCollection<BsonDocument> collection = database.getCollection(workload.getString("collection").getValue(),
-                BsonDocument.class);
-
-        JsonPoweredCrudTestHelper helper = new JsonPoweredCrudTestHelper("Workload executor", database, collection);
-
-        BsonArray operations = workload.getArray("operations");
-
-        outer:
-        while (!done) {
-            for (BsonValue cur : operations.getValues()) {
-                try {
-                    if (done) {
-                        break outer;
-                    }
-                    BsonDocument operation = cur.asDocument().clone();
-                    BsonValue expectedResult = operation.get("result");
-                    BsonDocument resultDocument = helper.getOperationResults(operation);
-
-                    if (expectedResult != null && !resultDocument.get("result").equals(expectedResult)) {
-                        numFailures.incrementAndGet();
-                    } else {
-                        numSuccesses.incrementAndGet();
-                    }
-                } catch (Exception e) {
-                    numErrors.incrementAndGet();
-                    e.printStackTrace();
-                }
+
+        System.out.println(fileDocument.toJson(JsonWriterSettings.builder().indent(true).build()));
```

Comment:
Can we remove these printlns?

Reply:
The println's are useful to understand what Astrolabe is asking the workload executor to do.  Since it's not production code, I don't think it matters whether we use a logger or println, as we only care that these show up when running the executor, and we don't mind that they always show up (we don't need runtime control as enabled by a logger).  So I think it's ok to leave as is, but let me know if you think there are other reasons to prefer a logger.

Reply:
I still think we should follow the same coding conventions for all our projects.

Reply:
Done